static/js/compose_actions.js

```diff
@@ -64,19 +64,35 @@ function get_focus_area(msg_type, opts) {
     }
 
     if (msg_type === "stream") {
-        return "#stream_message_recipient_stream";
+        return "#id_compose_select_stream";
     }
     return "#private_message_recipient";
 }
 
 // Export for testing
 export const _get_focus_area = get_focus_area;
 
+export function open_compose_stream_dropup() {
+    const $button = $("#id_compose_select_stream button");
+    if ($button) {
+        $button.trigger("click");
+    }
+}
+
+export function close_compose_stream_dropup() {
+    $("#id_compose_select_stream").removeClass("open");
+}
+
 export function set_focus(msg_type, opts) {
     const focus_area = get_focus_area(msg_type, opts);
     if (window.getSelection().toString() === "" || opts.trigger !== "message click") {
-        const $elt = $(focus_area);
-        $elt.trigger("focus").trigger("select");
+        if (focus_area === "#id_compose_select_stream") {
+            // Open the dropup menu instead of focusing on a field.
+            open_compose_stream_dropup();
+        } else {
+            const $elt = $(focus_area);
+            $elt.trigger("focus").trigger("select");
+        }
```

Comment:
Something I could do instead

```
export function set_focus(msg_type, opts) {
    if (window.getSelection().toString() === "" || opts.trigger !== "message click") {
        if (msg_type === "stream" && !opts.stream) {
            // Open the dropup menu instead of focusing on a field.
            open_compose_stream_dropup();
        } else {
            const focus_area = get_focus_area(msg_type, opts);
            const $elt = $(focus_area);
            $elt.trigger("focus").trigger("select");
        }
```

Reply:
The blueslip error is probably useful; I can easily see a bug where `get_focus_area` no longer works. 

Another idea, which might be cleaner, would be to make `get_focus_area` return a special value for the stream input ... or perhaps best, to write a `.select` handler that opens when that element is focused, and have this code remain unchanged.

As a side note, I noticed that using `tab` and `shift+tab` to navigate between widgets skips the stream input; I'm not entirely sure whether that issue is in this code, but it might be fixed by that last suggestion.

Reply:
I like that idea! I added a select handler. I'm not seeing the `tab` issue now, but let me know if you still have the issue.